### ISSUE TITLE
[feature request] preprocessor expressions

### ISSUE THREAD

**Issue by sebseb7** [michaelkamprath/bespokeasm#13]
At least low() and high() would be very useful for architectures with 16 bit address bus and 8 bit data bus.

```
sub ax,low(label)
sub ay,high(label)
```

[![image](https://user-images.githubusercontent.com/677956/175816801-67c52732-6767-49bc-89e9-2279e0445582.png)](http://ww1.microchip.com/downloads/en/DeviceDoc/40001917A.pdf#page17)

**Comment by michaelkamprath** [comment]
I have good news and better news.

The good news is that as of BespokeASM v0.2.2, you can do this:

```asm
sub ax,(label & $00FF)
sub ay,(label >> 8)
```

The better news is that I am thinking about adding syntax sugar to more concise doe the high byte / low byte operation for my own 8-bit data / 16-bit address project. I don't think I am going to make rich functions like the AVR assembler you quote, and `HIGH()`/`LOW()` doesn't seem that much more concise than the enable code I show above. A notation I has seen elsewhere is this:

```asm
sub ax,<label.    ; LSB
sub ay,>label.    ; MSB
```

Basically, less-than `<` is used to grab the least significant byte, and greater than `>` is used to grab the most significant byte. Then again the AVR assembly notation is much more explicit as to what it means. What are you thoughts on notation?

Of course, this all only applies to values that can be resolved at compile time. Run time determination requires an instruction or two.

**Comment by michaelkamprath** [comment]
I went ahead and picked a syntax. I went with `MSB()` and `LSB()`. An important note is that the `MSB()` function will get the most significant byte of the set of bits needs represent the number. So `MSB($12345)` would return `$01`.  Since I haven't implemented integer size hints yet, that means `MSB($00012345)` would also return `$01`. If you address labels are consistently 16-bit values, this will do what you want. I might also implement the `BYTEn()` functions like you highlight in the AVR assembly.

This feature will be available in the v0.2.3 release. you can try it right now on the `mincpu_support` branch.

**Comment by michaelkamprath** [comment]
I just realized the `MSB()` behavior I implemented will not work as you need if the address is something like `$0018`. I will change things to simply do byte slicing with a set of `BYTEn()` functions.
